src/test/rustdoc/intra-doc-link-generic-params.rs

```diff
@@ -0,0 +1,10 @@
+#![crate_name = "foo"]
+
+//! Here's a link to [`Vec<T>`] and one to [`Box<Vec<Option<T>>>`].
+//!
+//! And what about a link to just [`Option`] and, with the generic, [`Option<T>`]?
+
+// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html'
```

Comment:
Can you also check the text displayed is correct? Same for the other tests.
```suggestion
// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html"]' 'Vec<T>'
```